xdsl/ir.py

```diff
@@ -1003,6 +1007,10 @@ def __len__(self):
             result += 1
         return result
 
+    def __bool__(self) -> bool:
+        """Returns `True` if there are no operations in this block."""
+        return self.block.first_op is not None
```

Comment:
To me, this is one of the Python functionalities where I personally draw the line between nice, concise syntax and absolut madness, this being on the madness side obviously. I am in favor of explicit things, in this case in particular.

Now, I know this is exactly the kind of thing that Python users would describe as "pythonic", and if we want stay with the cool kids, we should design the framework with that in mind. Therefore, I am fine with adding this (with a sligthyly disgusted look on my face), as long as you let me keep the `is_empty` property. Then, the mad lads can use their fancy, confusing Python syntax, while me and the other saints can write proper code 😉

Reply:
Lol fine I'll add it back on `Block`. After Anton's comment, I substituted `assert False` for the implementation, and found a bunch of places where falseyness of `None` was used implicitly, with your name next to the lines in question, so feel free to consider yourself a Python cool kid :)

Reply:
I'm sure I only moved those around 😇